main.go

```diff
@@ -121,22 +121,28 @@ func buildChannels(usedTracks []m3u.Track) []LineupItem {
 
 	for _, track := range usedTracks {
 
-		var finalName string
+		var channName string
 		if track.TvgName == "" {
-			finalName = track.Name
+			channName = track.Name
 		} else {
-			finalName = track.TvgName
+			channName = track.TvgName
+		}
+
+		var channNum string
+		if track.TvgID == "" {
+			channNum = strconv.Itoa(gn)
+			gn += 1
+		} else {
+			channNum = track.TvgID
```

Comment:
Need to make sure Plex allows non-int strings as a channel number because in my m3u the Tvg ID is something like "BBCOne.UK"

Reply:
Indeed, i've only seen non-integer ones so far in my playlists as wel.
Testing will point out.